### ISSUE TITLE
In-place Rotation Failure

### ISSUE THREAD

**Issue by roborocklsm** [rpautrat/SuperPoint#94]
According to your paper, Super Point seems not stable for the in-place rotation situation. Do you have any idea why it happens? I believe you have tried to increase the ratio of in-place rotation image-pairs during training. Why the rotation transformation so different compared with others?

**Comment by rpautrat** [comment]
Hi,

This is not my paper, I only reimplemented the paper of Daniel DeTone, Tomasz Malisiewicz and Andrew Rabinovich in this repo.

The rotation itself is not so different from the other transformations, it's just that SuperPoint will fail for extreme in-plane rotations because it has not seen such transformations during training. I tried to increase the range of angles of in-plane rotation of the training images, but beyond some angle threshold, it has actually a negative effect on the repeatability score when the angles of rotation are too high.

It could help however to gradually increase the maximum amplitude of the rotation during training, so that the network learns progressively to tackle such transformations. I haven't tried this though.
